pkg/controllers/nodeclaim/disruption/consolidation.go

```diff
@@ -69,9 +70,13 @@ func (c *Consolidation) Reconcile(ctx context.Context, nodePool *v1.NodePool, no
 		return reconcile.Result{RequeueAfter: consolidatableTime.Sub(c.clock.Now())}, nil
 	}
 
-	// 6. Otherwise, add the consolidatable status condition
-	nodeClaim.StatusConditions().SetTrue(v1.ConditionTypeConsolidatable)
+	// 3. Otherwise, add the consolidatable status condition if not already set
 	if !hasConsolidatableCondition {
+		nodeClaim.StatusConditions().SetTrue(v1.ConditionTypeConsolidatable)
+		// We sleep here after a set operation since we want to ensure that we are able to read our own writes
+		// so that we avoid duplicating log lines due to quick re-queues from our node watcher
+		time.Sleep(100 * time.Millisecond)
```

Comment:
So, this sleep here isn't actually going to do anything. The sleep needs to come after the Patch to give the cache time to update. If we just sleep here without doing a cache update, we aren't really accomplishing the deduping that we intended

Reply:
i did change following how lifecycle controller does, you mentioned in the issue, but not sure does not duplicating logs